pkg/trustedcabundle/trustedcabundle.go

```diff
@@ -0,0 +1,153 @@
+package trustedcabundle
+
+import (
+	"context"
+	"strings"
+
+	corev1 "k8s.io/api/core/v1"
+	apierrs "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	dsci "github.com/opendatahub-io/opendatahub-operator/v2/apis/dscinitialization/v1"
+)
+
+const (
+	injectionOfCABundleAnnotatoion = "security.opendatahub.io/inject-trusted-ca-bundle"
+	CAConfigMapName                = "odh-trusted-ca-bundle"
+	CADataFieldName                = "odh-ca-bundle.crt"
+)
+
+func IsValidNamespace(ns client.Object) bool {
+	if !strings.HasPrefix(ns.GetName(), "openshift-") && !strings.HasPrefix(ns.GetName(), "kube-") &&
+		ns.GetName() != "default" && ns.GetName() != "openshift" && !HasCABundleAnnotationDisabled(ns) {
+		return true
+	}
+	return false
+}
+
+func HasCABundleAnnotationDisabled(ns client.Object) bool {
+	nsAnnotations := ns.GetAnnotations()
+
+	for annKey, annValue := range nsAnnotations {
+		if annKey == injectionOfCABundleAnnotatoion && annValue == "false" {
+			return true
+		}
+	}
+	return false
+}
+
+func AddCABundleConfigMapInAllNamespaces(ctx context.Context, cli client.Client, dscInit *dsci.DSCInitialization) error {
+	namespaceList := &corev1.NamespaceList{}
+	err := cli.List(ctx, namespaceList)
+	if err != nil {
+		return err
+	}
+
+	for _, ns := range namespaceList.Items {
+		if !strings.HasPrefix(ns.Name, "openshift-") && !strings.HasPrefix(ns.Name, "kube-") &&
+			ns.Name != "default" && ns.Name != "openshift" {
+			if err := CreateOdhTrustedCABundleConfigMap(ctx, cli, ns.Name, dscInit); err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+// createOdhTrustedCABundleConfigMap creates a configMap  'odh-trusted-ca-bundle' -- Certificates for the cluster
+// trusted CA Cert Bundle.
+func CreateOdhTrustedCABundleConfigMap(ctx context.Context, cli client.Client, namespace string, dscInit *dsci.DSCInitialization) error {
+	// Expected configmap for the given namespace
+	desiredConfigMap := &corev1.ConfigMap{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "ConfigMap",
+			APIVersion: "v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      CAConfigMapName,
+			Namespace: namespace,
+			Labels: map[string]string{
+				"app.kubernetes.io/part-of": "opendatahub-operator",
```

Comment:
can it be something e.g "app.kubernetes.io/part-of": "trusted-cabundle"
so it can be used for both upstream and downstream ?

Reply:
> > can it be something e.g "app.kubernetes.io/part-of": "trusted-cabundle"
> 
> But what would it mean? `trusted-cabundle` is not a higher-level app. At least that's how I would understand recommendations here https://kubernetes.io/docs/concepts/overview/working-with-objects/common-labels/#labels
> 
> Maybe we need a way to parametrize the operator name used in labels based on the distribution?

i could not think of something better than the original value of "opendatahub-operator"
on the other hand, resource like DSC and DSCI CR are labelled by bundle's alm-example or initialization-resource, If here we hardcode it, unless we have different code in downstream, it will be used this as you said for differnt distributions.

Reply:
Keeping  this as "opendatahub-operator" for now